swarm/api/http/server.go

```diff
@@ -644,10 +649,15 @@ func (s *Server) logError(format string, v ...interface{}) {
 
 func (s *Server) BadRequest(w http.ResponseWriter, r *Request, reason string) {
 	s.logDebug("bad request %s %s: %s", r.Method, r.uri, reason)
-	http.Error(w, reason, http.StatusBadRequest)
+	ShowError(w, &r.Request, fmt.Sprintf("Bad request %s %s: %s", r.Method, r.uri, reason), http.StatusBadRequest)
 }
 
 func (s *Server) Error(w http.ResponseWriter, r *Request, err error) {
 	s.logError("error serving %s %s: %s", r.Method, r.uri, err)
-	http.Error(w, err.Error(), http.StatusInternalServerError)
+	ShowError(w, &r.Request, fmt.Sprintf("Error serving %s %s: %s", r.Method, r.uri, err), http.StatusInternalServerError)
+}
+
+func (s *Server) NotFound(w http.ResponseWriter, r *Request, err error) {
+	s.logError("NOT FOUND error serving %s %s: %s", r.Method, r.uri, err)
```

Comment:
Given the server will probably get quite a few not found errors, I don't think it makes sense to log them all.